Library/Homebrew/service.rb

```diff
@@ -159,6 +159,26 @@ def requires_root?
       @require_root.present? && @require_root == true
     end
 
+    sig { params(value: T.nilable(T::Boolean)).returns(T.nilable(T::Boolean)) }
+    def run_at_load(value = nil)
+      case T.unsafe(value)
+      when nil
+        @run_at_load
+      when true, false
+        @run_at_load = value
+      else
+        raise TypeError, "Service#run_at_load expects a Boolean"
+      end
+    end
+
+    # Returns a `Boolean` describing if a service requires the command to execute at load.
+    # @return [Boolean]
+    sig { returns(T::Boolean) }
+    def run_at_load?
+      instance_eval(&@service_block)
```

Comment:
This re-evaluates on every call. Is that necessary?

Reply:
I'm just trying to obey the structure. Almost all of the methods for booleans in this file were set up like this. What exactly would you recommend me to change here?